dolphin/cmd/api.py

```diff
@@ -31,8 +31,24 @@
 from dolphin import service
 from dolphin import utils
 from dolphin import version
+import socket
 
 CONF = cfg.CONF
+LOG = log.getLogger(__name__)
+
+
+def is_port_in_use(address, port):
+    s = socket.socket()
+    LOG.debug("Trying to connect%s on port %s" % (address, port))
+    try:
```

Comment:
maybe you can simply use 
```
with socket.socket() as s:
   return s.connect_ex((address, port)
``` 
will remove all your try catch and closing the socket

Reply:
is this method work for ipv6 also?

Reply:
@kumarashit , done

Reply:
> is this method work for ipv6 also?

have not tested, but we are not aiming ipv6 support now..

Reply:
This code won't work for IPV6 as it will need the family as AF_INET6..this is default to AF_INET..BUt i think it's ok for now as IPV6 will need other considerations